src/http/Stream.cc

```diff
@@ -8,6 +8,7 @@
 
 #include "squid.h"
 #include "client_side_request.h"
+#include "clientStream.h"
```

Comment:
These and similar `#include` additions in this PR are necessary because client_side_request.h no longer includes clientStream.h (because it no longer needs names declared in clientStream.h). The affected .cc files were essentially abusing src/client_side_request.h to import clientStream.h for them.

P.S. This comment annotates this PR but does not request any changes.

Reply:
> The affected .cc files were essentially abusing src/client_side_request.h to import clientStream.h for them.

FTR; having dependency B include dependency A, is **not** "abuse" when dependency B and the .cc both depend on A. Such include reduction is **good* practice as the circular dependency problems caused by overly dense includes are worse to resolve than easily identified "missing include" build errors.

Reply:
> > The affected .cc files were essentially abusing src/client_side_request.h to import clientStream.h for them.
> 
> having dependency B include dependency A, is **not** "abuse" when dependency B and the .cc both depend on A. 

That is correct but irrelevant: What my comment described (and what is present in the current official code) is "abuse" (for the lack of a better word). Your comment describes a different kind of inclusion. I will try to clarify the abuse case using a distilled example below.

Abuse pattern has at least three source files:

```c++
// This is A.h. It defines class A.
class A {};
```

```c++
// This is B.h. It currently uses class A.
#include "A.h"
class B: A {};
```

```c++
// This is C.cc. It uses both A and B classes (independently).
#include "B.h"
class C1: A {}; // XXX: C.cc does not include A.h (but code currently "works").
class C2: B {}; // OK: C.cc includes B.h.
```

To fix abuse, C.cc should be written like this instead:

```c++
// This is C.cc. It uses both A and B classes (independently).
#include "A.h"
#include "B.h"
class C1: A {};
class C2: B {};
```

N.B. The above also holds true if we replace `C.cc` with `C.h` -- the top-level file nature does not matter here. My comment was about `.cc` so I am using `C.cc` in this example.

> the circular dependency problems caused by overly dense includes are worse to resolve than easily identified "missing include" build errors.

In this context, non-abusive variations do not cause circular dependency problems. They do not introduce any new dependencies.